include/miopen/miopen.h

```diff
@@ -583,6 +583,50 @@ typedef enum
  */
 MIOPEN_EXPORT miopenStatus_t miopenCreateTensorDescriptor(miopenTensorDescriptor_t* tensorDesc);
 
+/*! @brief Set the quantization scale of Tensor Descriptor
+ *
+ * API for setting the quantization scale of tensor descriptor.
+ * @param tensorDesc  Pointer to a tensor descriptor type
+ * @param quantScales Quantization scales
+ * @param len         Length of quantization scales buffer
+ * @return            miopenStatus_t
+ */
+MIOPEN_EXPORT miopenStatus_t miopenSetQuantizationScales(miopenTensorDescriptor_t tensorDesc,
+                                                         double* quantScales,
```

Comment:
The impression I get for the element-wise operation is that we use the following equation to compute the result elements:

`int8_res = (int32_res / scale) + bias`

However, due the the overhead of division in gpu, I'd like us to be passed with a multiplication for scaling factor, such that:

`int8_res = (int32_res * new_scale) + bias`

Now, does framework guarantee that they will do the conversion of `new_scale = 1 / scale` when the pass this argument to MIOpen? If we don't have that guarantee, this needs to carry out the division right in the API.

Reply:
I recommend renaming quantScales to quantizationMultiplierScales or quantizationFactorScales. The scaling operation must be also described, e.g. like this:
```
tensor[n,k,h,w] = quantMultiplierScales[k] * tensor[n,k,h,w] + quantBias[k]
```
This is expected to resolve some abmiquities mentioned at https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1625#discussion_r1095897476

Reply:
BTW how does the the library know how many elements from `quantScales` should be used? This is necessary to copy data from CPU to GPU.

If this is hardwired to `C`, then the API becomes rigid in terms of future extension (but the answer to https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1625/files#r1128369353 becomes very simple: `K`).

I would prefer flexibility. Let's add another parameter, `numQuantScales`.

This would also work for output tensors. There is no need to initialize them. For output  tensors the user can provide NULL as `quantScales` and the library will skip initialization (copying) -- it will simply reserve space for output scales.